### ISSUE TITLE
Merge dev branch into master - post refactor

### ISSUE THREAD

**Issue by jarmoza** [neurobagel/annotation_tool#435]
Now that we are complete with the store refactor, subsequent features and testing, I will no merge the development branch `dev_components_talk_to_store_directly` into the `master` branch of the annotation tool.

**Comment by surchs** [comment]
:+1: Only comment: this may be a good candidate for pair programming - if there are any merge conflicts to be resolved, that is (but I strongly expect there will be).

**Comment by jarmoza** [comment]
See #436 WIP PR. So far so good actually with merges with the exception of something quirky happening on GitHub's side re: e2e testing.

Re-running now here: https://github.com/neurobagel/annotation_tool/actions/runs/5137827457

If this doesn't resolve will reach out to channel tomorrow for 2 sets of eyes.

**Comment by surchs** [comment]
hhm. e2e error could be many things. seems like the dev build fails which would make e2e fail as well. what are the merge conflicts? I guess you only see those once you take the PR out of draft mode?

**Comment by surchs** [comment]
this seems clear enough to work on to me (and I think it is already worked on, no?). Either way: let's move to specify-done (or clarify the remaining things) and then get it worked on

**Comment by surchs** [comment]
Additional note on https://github.com/neurobagel/annotation_tool/pull/437 (that is failing atm): I'm pretty sure the failure is because we introduced https://github.com/neurobagel/annotation_tool/blob/7ca5712ad9ecab8c7b9a0a0832304129538c7e49/nuxt.config.js#L64 in https://github.com/neurobagel/annotation_tool/commit/095261b2d8050c8e672d11ab99ff263e9f525695.

I made a little test to remove the "static" build target in https://github.com/neurobagel/annotation_tool/pull/438 and indeed the e2e passes (sidenote: still haven't figured out how to run GH actions easily on a branch so I don't have to make a PR, I guess manual would work?). 

Unfortunately just removing "static" isn't the solution either, because we need that to deploy on GH pages. I could imagine that a fix would be to temporarily override the build target (i.e. make it "not static") in the GH actions workflow for the e2e test. Or the alternative is probably to figure out how to run the e2e tests with cypress against a statically compiled version of the app - I strongly believe that the latter is going to be harder.
